code/obj/item/device/pda2/smallprogs.dm

```diff
@@ -513,80 +513,85 @@ Code:
-				stuff += "Pressure Inlet: [round(MIXTURE_PRESSURE(circ2.air1), 0.1)] kPa  Outlet: [round(MIXTURE_PRESSURE(circ2.air2), 0.1)] kPa<BR>"
-				stuff += "<BR>"
+				. += "<B>Cold Loop</B><BR>"
+				. += "Temperature Inlet: [round(circ2.air1?.temperature, 0.1)] K  Outlet: [round(circ2.air2?.temperature, 0.1)] K<BR>"
+				. += "Pressure Inlet: [round(MIXTURE_PRESSURE(circ2?.air1), 0.1)] kPa  Outlet: [round(MIXTURE_PRESSURE(circ2?.air2), 0.1)] kPa<BR>"
+				. += "<BR>"
 
-		if (length(collector_controlers))
+		if(length(by_type[/obj/machinery/power/collector_control]))
 			var/controler_index = 1
 			var/collector_index = 1
-			for(var/obj/machinery/power/collector_control/C as() in collector_controlers)
+			for_by_tcl(C, /obj/machinery/power/collector_control)
 				collector_index = 1
-				if(C?.active)
+				if(C?.active && C.z == 1)
 					engine_found = TRUE
-					stuff += "<BR><h4>Radiation Collector [controler_index++] Status</h4>"
-					stuff += "Output: [engineering_notation(C.lastpower)]W<BR>"
-					if(C.CA1?.active) stuff += "Collector [collector_index++]: Tank Pressure: [C.P1 ? round(MIXTURE_PRESSURE(C.P1?.air_contents), 0.1) : "ERR"] kPa<BR>"
-					if(C.CA2?.active) stuff += "Collector [collector_index++]: Tank Pressure: [C.P2 ? round(MIXTURE_PRESSURE(C.P2?.air_contents), 0.1) : "ERR"] kPa<BR>"
-					if(C.CA3?.active) stuff += "Collector [collector_index++]: Tank Pressure: [C.P3 ? round(MIXTURE_PRESSURE(C.P3?.air_contents), 0.1) : "ERR"] kPa<BR>"
-					if(C.CA4?.active) stuff += "Collector [collector_index++]: Tank Pressure: [C.P4 ? round(MIXTURE_PRESSURE(C.P4?.air_contents), 0.1) : "ERR"] kPa<BR>"
-					stuff += "<BR>"
-
-		if (!engine_found)
-			stuff += "<BR><B>Error!</B> No power source detected!<BR><BR>"
-
-		stuff += "<HR>"
-		if (laser)
-			stuff += "<BR><B>Power Transmition Laser Status</B><BR>"
-			stuff += "Currently Active: "
-
-			if(laser.firing)
-				stuff += "Yes<BR>"
-			else
-				stuff += "No<BR>"
-
-			stuff += "Power Stored: [engineering_notation(laser.charge)]J ([round(100.0*laser.charge/laser.capacity, 0.1)]%)<BR>"
-			stuff += "Power Input: [engineering_notation(laser.chargelevel)]W<BR>"
-			stuff += "Power Output: [engineering_notation(laser.output)]W<BR>"
+					. += "<BR><h4>Radiation Collector [controler_index++] Status</h4>"
+					. += "Output: [engineering_notation(C.lastpower)]W<BR>"
+					if(C.CA1?.active) . += "Collector [collector_index++]: Tank Pressure: [C.P1 ? round(MIXTURE_PRESSURE(C.P1?.air_contents), 0.1) : "ERR"] kPa<BR>"
+					if(C.CA2?.active) . += "Collector [collector_index++]: Tank Pressure: [C.P2 ? round(MIXTURE_PRESSURE(C.P2?.air_contents), 0.1) : "ERR"] kPa<BR>"
+					if(C.CA3?.active) . += "Collector [collector_index++]: Tank Pressure: [C.P3 ? round(MIXTURE_PRESSURE(C.P3?.air_contents), 0.1) : "ERR"] kPa<BR>"
+					if(C.CA4?.active) . += "Collector [collector_index++]: Tank Pressure: [C.P4 ? round(MIXTURE_PRESSURE(C.P4?.air_contents), 0.1) : "ERR"] kPa<BR>"
```

Comment:
you don't need the inner ?.s on the PX vars

Reply:
Removed redundancy.